tool/common/common.go

```diff
@@ -30,6 +31,17 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+// ExitCodeError wraps an exit code as an error.
+type ExitCodeError struct {
+	// Code is the exit code
+	Code int
+}
+
+// Error implements the error interface.
+func (e *ExitCodeError) Error() string {
```

Comment:
Should we pass `ExitCodeError` by value?

Reply:
No I don't think so, both because the exitCodeError that was originally defined in tsh.go used a pointer receiver to implement `error` and because from what I've found online it's best to use a pointer receiver. 

The semantics of error checking are more complicated (and mistake-prone) using a value receiver, because the underlying error may or may not be a pointer and may or may not be equal to another error depending on whether the underlying type is a value or a pointer - this means error checking code needs to consider both cases and handle appropriately. `errors.As` in particular has behavior I find to be too subtle depending on the underlying type. See the difference here:
https://go.dev/play/p/OzdcZal12OC

Reply:
It's not so common for code to be concerned about the underlying type of interfaces in general, but in the case of the `error` interface, it's common for code to inspect the underlying type of an `error` to decide how to handle it.